miniapps/dfem/dfem-minimal-surface.cpp

```diff
@@ -179,12 +179,9 @@ class MinimalSurface : public Operator
 
          dres_du->Mult(z, y);
 
-         auto d_y = y.HostReadWrite();
-         const auto d_x = x.HostRead();
-         for (int i = 0; i < minsurface->ess_tdofs.Size(); i++)
-         {
-            d_y[minsurface->ess_tdofs[i]] = d_x[minsurface->ess_tdofs[i]];
-         }
+         // Reuse z as a temporary vector to avoid unnecessary allocations.
+         x.GetSubVector(minsurface->ess_tdofs, z);
+         y.SetSubVector(minsurface->ess_tdofs, z);
```

Comment:
I'm not sure this makes sense -- code that can easily be converted to one `mfem::forall` kernel is instead replaced by two kernels.